src/data/restore.ts

```diff
@@ -26,96 +20,105 @@ const getFontFamilyByName = (fontFamilyName: string): FontFamily => {
   return DEFAULT_FONT_FAMILY;
 };
 
+function migrateElementWithProperties<T extends ExcalidrawElement>(
+  element: T,
+  extra: Omit<T, keyof ExcalidrawElement>,
+): T {
+  const base: Pick<T, keyof ExcalidrawElement> = {
+    type: element.type,
+    // all elements must have version > 0 so getDrawingVersion() will pick up
+    //  newly added elements
+    version: element.version || 1,
+    versionNonce: element.versionNonce ?? 0,
+    isDeleted: false,
+    id: element.id || randomId(),
+    fillStyle: element.fillStyle || "hachure",
+    strokeWidth: element.strokeWidth || 1,
+    strokeStyle: element.strokeStyle ?? "solid",
+    roughness: element.roughness ?? 1,
+    opacity: element.opacity == null ? 100 : element.opacity,
+    angle: element.angle || 0,
+    x: element.x || 0,
+    y: element.y || 0,
+    strokeColor: element.strokeColor,
+    backgroundColor: element.backgroundColor,
+    width: element.width || 0,
+    height: element.height || 0,
+    seed: element.seed ?? 1,
+    groupIds: element.groupIds || [],
+  };
+
+  return {
+    ...base,
+    ...getNormalizedDimensions(base),
+    ...extra,
+  } as T;
```

Comment:
I'm asserting here because TS cannot understand that merging `Pick<T, keyof ExcalidrawElement>` and `Omit<T, keyof ExcalidrawElement>` gets us `T` back.